### ISSUE TITLE
Date, DateTime and Time format on export/API

### ISSUE THREAD

**Issue by PhilippGrashoff** [atk4/data#403]
Hi,
as just discussed in the weekly meeting, the ATK API shouldn't export Date, DateTime and Time fields as serialized \DateTime objects, but as strings like (php spreaking):
Datetime: ISO 8601
Date: YYYY-mm-dd
Time: HH:ii:ss

The question open is: Should this be the case on API level only, or deeper in ATK like in export() function itself?

**Comment by PhilippGrashoff** [comment]
Related: #394

**Comment by PhilippGrashoff** [comment]
Regarding API, changing `Model->export()` wouldnt be enough, as GETting a single record, POST and PATCH use `Model->get()`, (without param), which creates a one-dimensional array.

**Comment by PhilippGrashoff** [comment]
HI there,
as there is the new Persistence ArrayOfStrings now, maybe the simplest way for API is to run all values through ArrayOfStrings typecast, which should take good care of at least all date related fields. What do you think?

**Comment by DarkSide666** [comment]
Yes probably, but I think that's a bit to much work to involve another persistence.

I think it's not yet documented, but now we have this:
* `Model->getField()` - useful method to return Field objects. It's better to use this instead of getElement(). It's not related to this issue, but I just wanted to mention that :)
* `Field->toString()` - returns field value as string no matter what type field is. You can also pass value parameter to this method and get it typecasted to string.
* `Model->export($fields, $key_field, FALSE)` - set 3rd parameter to false and export will not typecast values. This is useful to see what values you're actually storing in database for example.

**Comment by mvorisek** [comment]
@PhilippGrashoff please check if fixed, I belive if you export as DB/unmanaged values, all values are casted correctly
